### PR TITLE
Fix Levenstein distance on empty buffers ##radiff2

### PR DESCRIPTION
**Pull request by swoops**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [x] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

I saw issue #17237 and took a look. I could not get the unit test to run but I spotted an unhanded edge case which the test uses. This PR will fix the case when one or both provided buffers are of length 0.

**Test plan**

I couldn't get the unit tests to run, but this works now:
```
radiff2 -ss /tmp/tests/1/*
File size differs 5 vs 0
similarity: 0.000
distance: 5
```

Previously distance was reported as 0.

**Closing issues**

Maybe #17237 ?

### CODE REVIEW

**Comment by ret2libc** [comment]
Just to understand: were you able to run the test but not to make the Levenstein commented code work or weren't you able to run the test at all?

File: libr/util/udiff.c [hunk 1]
```diff
@@ -239,11 +239,22 @@ R_API bool r_diff_buffers_distance_levenstein(RDiff *d, const ut8 *a, ut32 la, c
 	stop = bLen;
 	// Preliminary tests
 
-	//Do we have both files a & b, and are they at least one byte?
-	if (!aBufPtr || !bBufPtr || aLen < 1 || bLen < 1) {
+	//Do we have both files a & b
+	if (!aBufPtr || !bBufPtr) {
```
Comment (trufae) [inline libr/util/udiff.c:243]: uhm, it is valid to pass those nulls? shouldnt be asserting because of bad api usage?

**Comment by swoops** [comment]
> Just to understand: were you able to run the test but not to make the Levenstein commented code work or weren't you able to run the test at all?

I don't think I got the tests to run at all. I ran `r2r unit` from the test directory because of a comment seen in `run_unit.sh` but output seemed to indicate I was not doing things correctly. I was excited to learn the diff algorithm anyways, so I went looking at that instead of r2r code. I spotted the missed edge case for when a zero length buffer. So I figured I would send a quick fix. If the test is only failing on `R("", "zzz", 3, 3)` and `R("meow", "", 4, 4)` then this PR should fix it. This code should also cover the untested `R("", "", 0, 0)` case.


> Fix the PR title, follow the rules described in #17335
I updated the PR title. If someone can confirm this is the fix for  #17237 then I will update the PR title to "Fix #17237 zero length buffer ##radiff2" or the like. Sorry my PR titles are always terrible, I appreciate the direction in  #17335.

File: libr/util/udiff.c (hunk 1 above)
Comment (swoops) [reply libr/util/udiff.c]: I agree, I updated to check that at function entry with `r_return_val_if_fail`.

**Comment by ret2libc** [comment]
> I don't think I got the tests to run at all. I ran `r2r unit` from the test directory because of a comment seen in `run_unit.sh` but output seemed to indicate I was not doing things correctly.

I think if you do `cd test/unit; make` you should be able to compile the unit test. Then just run `./bin/test_diff`.
